### PR TITLE
major reorg of project for containers

### PR DESCRIPTION
**Pull request by shaikh-rashid**
Renaming, refactoring, reordering, re...everything. `Listener.py` has been left alone for now but the world it once knew is gone. This PR focuses on container-driven development. A single `docker-compose.yml` represents the test environment and the external dependencies (ie. apps) are also running as containers. You can also use `docker-compose build` to create the listener container. `.env` provides single file to control all tunable global environment variables for the system state. More changes to `listener.py` to come in future PRs.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM python:3.6.0
-COPY . .
+FROM python:3.7.3
 RUN pip install flask
 RUN pip install requests
 RUN pip install minio
 RUN pip install wes-service
-CMD [ "python", "./Listener.py"]
+COPY listener.py /srv/server.py
```
Comment (shaikh-rashid) [inline Dockerfile]: @SamarthVP take a look here: it is better to run COPY commands last as they will always cause the container to change. If the RUN steps do not change (ie. dependencies do not change/update), Docker can used a cached version of the container image to rebuild the container, saving us time. By moving COPY near the end, we avoid having to re-run these RUN steps!

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,159 @@
+### macOS
```
Comment (shaikh-rashid) [inline .gitignore]: Using `.gitignore` is always recommended, it helps avoid accidentally adding/commiting files we don't want to be a part of our repo. `.gitignore` is project/repo specific, while `.gitignore_global` is obviously for all git projects.

File: .env [hunk 3]
```diff
@@ -0,0 +1,16 @@
+MINIO_VERSION=latest
+MINIO_UI_PORT=9000
+MINIO_REGION=us-east-1
+MINIO_DOMAIN=candig.local
+MINIO_DATA_DIR=/data
+
+MINIO_ACCESS_KEY=miniotest
```
Comment (shaikh-rashid) [inline .env]: @SamarthVP it is bad development practice to add secrets into repos, however in this case we are simply building a test case to prove the concept is viable. Before reaching production, these sorts of secrets are removed...

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  minio:
+    image: minio/minio:${MINIO_VERSION:-latest}
+    volumes:
+      - minio-data:/data
+      - minio-config:/root/.minio
+      - minio-events:/tmp/events
+    ports:
+      - "${MINIO_UI_PORT:-9000}:9000"
+    environment:
+      - MINIO_REGION=${MINIO_REGION:-us-east-1}
```
Comment (shaikh-rashid) [inline docker-compose.yml]: @SamarthVP all these `${<var>:-<default>}` are an easy way to be able to quickly change different parameters of the `docker-compose.yml` without having to rewrite. `docker-compose` will read `.env` automatically, so we can use this feature to set "custom" global environments variables for our test case

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,66 @@
+addict==2.2.0
```
Comment (shaikh-rashid) [inline requirements.txt]: @SamarthVP capture requirements with `pip freeze --local > requirements.txt`

File: Dockerfile (hunk 1 above)
Comment (SamarthVP) [reply Dockerfile]: Oh yeah, that's smart

File: requirements.txt (hunk 5 above)
Comment (SamarthVP) [reply requirements.txt]: That's very useful

**Pull request merged**